packages/xo-web/src/common/xo-compression/select-compression.js

```diff
@@ -0,0 +1,65 @@
+import _ from 'intl'
+import PropTypes from 'prop-types'
+import React from 'react'
+import { injectState, provideState } from 'reaclette'
+import { omit } from 'lodash'
+
+import decorate from '../apply-decorators'
+import Select from '../form/select'
+import { connectStore } from '../utils'
+import { createGetObject } from '../selectors'
+
+import { isZstdSupported } from '.'
+
+const COMMON_OPTIONS = [
+  {
+    label: _('stateDisabled'),
+    value: '',
+  },
+  {
+    label: _('chooseCompressionGzipOption'),
+    value: 'native',
+  },
+]
+
+const ZSTD_OPTION = {
+  label: _('chooseCompressionZstdOption'),
+  value: 'zstd',
+}
+
+const SELECT_COMPRESSION_PROP_TYPES = {
+  container: PropTypes.string,
+  onChange: PropTypes.func.isRequired,
+  value: PropTypes.string.isRequired,
+}
+
+const SelectCompression = decorate([
+  connectStore(() => ({
+    container: createGetObject((_, { container }) => container),
+  })),
+  provideState({
+    computed: {
+      options: (_, { container }) =>
+        container === undefined || isZstdSupported(container)
+          ? [...COMMON_OPTIONS, ZSTD_OPTION]
+          : COMMON_OPTIONS,
+      selectProps: (_, props) =>
+        omit(props, Object.keys(SELECT_COMPRESSION_PROP_TYPES)),
+    },
+  }),
+  injectState,
+  ({ onChange, state, value }) => (
+    <Select
+      labelKey='label'
+      onChange={onChange}
```

Comment:
Why do you remove `onChange` &` value` from `props` and add them again?

Reply:
I removed props related to the component `SelectCompression`.

Reply:
and you put the same, you don't recompute them, right?

Reply:
I don't understand the question. Do you talk about https://github.com/xapi-project/xen-api/commit/adca3b750da06deb9a141ad2447605601c49a177 ?

Reply:
You remove `onChange` and` value` from `props` to calculate` selectProps` and add them here, right? So don't remove them.
That's all. ^_^

Reply:
I don't remove `onChange` and `value` from `props`. A create a computed which only contains props not related to `selectCompression`.

Reply:
I don't understand why?
First: computing `selectProps` from `props`  (We can say removing `onChange/value/container` from `props` of `SelectCompression`)
Second:  adding `selectProps`, `onChange` and `value` to `Select` props.
So you want only remove `container` from `props` of `SelectCompression`.

Reply:
I'll delete `onChange` and `value` from the definition because it's not used by `SelectCompression` and it'll respond to your request.